### PR TITLE
fix: Added overlap translate to XYAxis svgData

### LINKED ISSUE #3126: ChartPanel with Column Bars Hovering
If you run this example:

```
import { Column } from "@hpcc-js/chart";
import { ChartPanel } from "@hpcc-js/layout";

const cp = new ChartPanel()
    .widget(new Column())
    .columns(["Subject", "Year 1", "Year 2", "Year 3"])
    .data([
        ["Jan", 1, 6, 21],
		["Feb", 0, 6, 17],
		["Mar", 0, 1, 10],
		["Apr", 0, 3, 6],
		["May", 1, 2, 13]
    ])
    .target("target")
    .title("Exam Results 2008->10")
    .legendVisible(true)
    .render();
```

you can observe the behavior.  Basically the bars are rendering in a way that is hovering over the x-axis.  Not sure why.  Please fix ASAP.

Screenshot:

image

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #3126

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised new issues to address them separately

## Testing:
I've tested the widgets that extend XYAxis in a standalone html page.

### CODE REVIEW

**Comment by jbrundage** [comment]
image

File: packages/chart/src/Axis.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ export class Axis extends SVGWidget {
     protected svg;
     protected svgAxis;
     protected svgGuides;
+    _overlap;
```
Comment (GordonSmith) [inline packages/chart/src/Axis.ts]: Should be protected

File: packages/chart/src/XYAxis.ts [hunk 2]
```diff
@@ -389,13 +389,14 @@ export class XYAxis extends SVGWidget {
             .tickLength(this.yAxisGuideLines() ? maxCurrExtent : 0)
             .render()
             ;
-
         this.svgDataClipRect
-            .attr("width", width)
-            .attr("height", height)
+            .attr("width", width - (isHorizontal ? this.xAxis._overlap.left : -this.yAxis._overlap.top))
+            .attr("height", height - (isHorizontal ? this.yAxis._overlap.top : -this.xAxis._overlap.left))
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: This code should be moved into "calcMargin" and if possible, should be completely encapsulated inside Axis

File: packages/chart/src/XYAxis.ts [hunk 3]
```diff
@@ -389,13 +389,14 @@ export class XYAxis extends SVGWidget {
...
             ;
+        const transX = isHorizontal ? this.xAxis._overlap.left : -this.yAxis._overlap.top;
+        const transY = isHorizontal ? this.yAxis._overlap.top : -this.xAxis._overlap.left;
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: This code should be moved into "calcMargin" and if possible, should be completely encapsulated inside Axis

File: packages/chart/src/XYAxis.ts [hunk 4]
```diff
@@ -389,13 +389,14 @@ export class XYAxis extends SVGWidget {
...
         this.svgData.transition()
-            .attr("transform", "translate(" + this.margin.left + "," + this.margin.top + ")")
+            .attr("transform", "translate(" + (this.margin.left + transX) + "," + (this.margin.top + transY) + ")")
```
Comment (GordonSmith) [inline packages/chart/src/XYAxis.ts]: This code should be moved into "calcMargin" and if possible, should be completely encapsulated inside Axis

**Comment by GordonSmith** [comment]
Closing in favor of:  https://github.com/hpcc-systems/Visualization/pull/3129

**Pull request closed without merging**